embassy-stm32/src/exti/blocking.rs

```diff
@@ -0,0 +1,161 @@
+//! Blocking EXTI (External Interrupt) functionality
+//!
+//! This module provides a non-async version of the EXTI input driver designed
+//! for use with custom interrupt handlers or frameworks like RTIC.
+use super::low_level::{self, TriggerEdge};
+use super::Channel;
+use crate::gpio::{Input, Level, Pin as GpioPin, Pull};
+use crate::Peri;
+
+/// EXTI input driver for custom interrupt handlers or frameworks like RTIC.
+///
+/// This driver augments a GPIO `Input` with EXTI functionality but is designed
+/// for use with interrupt-driven frameworks rather than async/await patterns.
+pub struct ExtiInput<'d> {
+    pin: Input<'d>,
+}
+
+impl<'d> Unpin for ExtiInput<'d> {}
+
+impl<'d> ExtiInput<'d> {
+    /// Creates a new EXTI input for interrupt-driven usage
+    ///
+    /// This function creates a new EXTI input that can be used with custom interrupt handlers or
+    /// interrupt-driven frameworks like RTIC. It initializes the GPIO pin with the specified pull
+    /// configuration, sets up the EXTI channel for the pin, and enables the interrupt.
+    ///
+    /// # Arguments
+    /// * `pin` - The GPIO pin to use
+    /// * `ch` - The EXTI channel corresponding to the pin
+    /// * `pull` - The pull configuration for the pin
+    /// * `trigger_edge` - The edge triggering mode (falling, rising, or both)
+    ///
+    /// # Returns
+    /// A new `ExtiInput` instance with interrupts enabled
+    pub fn new<T: GpioPin>(
+        pin: Peri<'d, T>,
+        ch: Peri<'d, T::ExtiChannel>,
+        pull: Pull,
+        trigger_edge: TriggerEdge,
+    ) -> Self {
+        // Needed if using AnyPin+AnyChannel.
+        assert_eq!(pin.pin(), ch.number());
+
+        let pin_num = pin.pin();
+        let port = pin.port();
+
+        low_level::configure_and_enable_exti(pin_num, port, trigger_edge);
+
+        Self {
+            pin: Input::new(pin, pull),
+        }
+    }
+
+    /// Configures the EXTI line to detect rising edges
+    ///
+    /// After calling this method, the EXTI line will generate interrupts
+    /// on rising edges of the input signal.
+    ///
+    /// # Returns
+    /// A mutable reference to self for method chaining
+    pub fn rising_edge(&mut self) -> &mut Self {
```

Comment:
don't make "builder-like" APIs that do chaining, it's inconisnstent with how the rest of the Embassy APIs look like.